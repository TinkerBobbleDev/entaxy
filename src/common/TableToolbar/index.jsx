import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import { lighten } from '@material-ui/core/styles/colorManipulator'

const styles = theme => ({
  root: {
    paddingRight: theme.spacing(1),
    paddingLeft: '8px',
    margin: '10px'
  },
  highlight: {
    color: theme.palette.secondary.main,
    backgroundColor: lighten(theme.palette.secondary.light, 0.85)
  },
  spacer: {
    flex: '1 1 100%'
  },
  actions: {
    color: theme.palette.text.secondary
  },
  title: {
    flex: '0 0 auto'
  }
})

const TableToolbar = ({
  title,
  subTitle,
  selectedItems,
  classes,
  children
}) => (
  <Toolbar
    className={classNames(classes.root, {
      [classes.highlight]: selectedItems.length > 0
    })}
  >
    <div className={classes.title}>
      {selectedItems.length > 0 ? (
        <Typography color="inherit" variant="subtitle1">
          {selectedItems.length}
          &nbsp;selected
        </Typography>
      ) : (
        <div>
          <Typography variant="h6">{title}</Typography>
          <Typography variant="caption">{subTitle}</Typography>
        </div>
      )}
    </div>
    <div className={classes.spacer} />
    <div className={classes.actions}>
      {children}
    </div>
  </Toolbar>
)

TableToolbar.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.node,
  selectedItems: PropTypes.array.isRequired,
  classes: PropTypes.object.isRequired,
  children: PropTypes.node
}

TableToolbar.defaultProps = {
  title: undefined,
  subTitle: undefined,
  children: undefined
}

export default withStyles(styles)(TableToolbar)
