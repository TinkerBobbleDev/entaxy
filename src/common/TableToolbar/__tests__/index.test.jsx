import React from 'react'
import renderer from 'react-test-renderer'
import TableToolbar from '../'

describe('TableToolbar', () => {
  it('matches snapshot with some selected items', () => {
    const component = renderer.create((
      <TableToolbar
        title="Transactions"
        selectedItems={['a']}
      >
        <div>Content 1</div>
        <div>Content 2</div>
      </TableToolbar>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })


  it('matches snapshot with no selected items', () => {
    const component = renderer.create((
      <TableToolbar
        title="Transactions"
        subtitle="Some error was found"
        selectedItems={[]}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
