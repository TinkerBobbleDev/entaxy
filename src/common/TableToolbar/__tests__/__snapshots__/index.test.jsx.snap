// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableToolbar matches snapshot with no selected items 1`] = `
<div
  className="MuiToolbar-root-6 MuiToolbar-regular-8 MuiToolbar-gutters-7 TableToolbar-root-1"
>
  <div
    className="TableToolbar-title-5"
  >
    <div>
      <h6
        className="MuiTypography-root-10 MuiTypography-h6-27"
      >
        Transactions
      </h6>
      <span
        className="MuiTypography-root-10 MuiTypography-caption-20"
      />
    </div>
  </div>
  <div
    className="TableToolbar-spacer-3"
  />
  <div
    className="TableToolbar-actions-4"
  />
</div>
`;

exports[`TableToolbar matches snapshot with some selected items 1`] = `
<div
  className="MuiToolbar-root-6 MuiToolbar-regular-8 MuiToolbar-gutters-7 TableToolbar-root-1 TableToolbar-highlight-2"
>
  <div
    className="TableToolbar-title-5"
  >
    <h6
      className="MuiTypography-root-10 MuiTypography-subtitle1-28 MuiTypography-colorInherit-39"
    >
      1
       selected
    </h6>
  </div>
  <div
    className="TableToolbar-spacer-3"
  />
  <div
    className="TableToolbar-actions-4"
  >
    <div>
      Content 1
    </div>
    <div>
      Content 2
    </div>
  </div>
</div>
`;
